From f1ecf54f8190351b6b1c2f163375405c4b238f64 Mon Sep 17 00:00:00 2001
From: Adrian Ratiu <adrian.ratiu@collabora.com>
Date: Fri, 12 Apr 2024 17:42:38 +0300
Subject: [PATCH] libtypec_sysfs_ops: define feature test macro for nftw
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

According to the man page [1], the definiton of nftw() is
hidden behind the _XOPEN_SOURCE 500 feature test, so we
must enable it to avoid warnings or errors (if the toolchain
is configured to treat them as errors) like this:

/var/tmp/portage/dev-libs/libtypec-0.5.0/work/libtypec-0.5.0-Source/libtypec_sysfs_ops.c:1059:13:
warning: implicit declaration of function ‘nftw’; did you mean ‘ftw’?
[-Wimplicit-function-declaration]
 1059 |         if (nftw ("/dev/bus/usb/", count_billbrd_if, fd_limit, 0) != 0)
      |             ^~~~
      |             ftw

[1] https://linux.die.net/man/3/nftw
Upstream-Status: Backport [upstream commit f1ecf54]
---
 libtypec_sysfs_ops.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libtypec_sysfs_ops.c b/libtypec_sysfs_ops.c
index 747b562..bfb5246 100644
--- a/libtypec_sysfs_ops.c
+++ b/libtypec_sysfs_ops.c
@@ -29,6 +29,11 @@ SOFTWARE.
  * @brief Functions for libtypec sysfs based operations
  */
 
+/**
+ *  required for enalbing nftw(), which is part of SUSv1.
+ */
+#define _XOPEN_SOURCE 500
+
 #include "libtypec_ops.h"
 #include <dirent.h>
 #include <stdio.h>
-- 
2.43.2

